_search-plugins/ubi/ubi-javascript-collector.md

```diff
@@ -0,0 +1,118 @@
+---
+layout: default
+title: UBI JavaScript Collector
+parent: User Behavior Insights
+has_children: false
+nav_order: 10
+---
+
+# UBI JavaScript collector
+
+UBI comes with a very basic JavaScript client that manages the life cycle of the `query_id` for a specific search and can create UBI Event data structures and store them for specific actions.
+
+For more information about the schema, see [UBI index schemas]({{site.url}}{{site.baseurl}}/search-plugins/ubi/schemas/).
+
+We recommend that you refer to the client as a starting point for your own specific needs.
+
+## Installation
+
+The client comes as a single file `ubi.js` and only has a dependency on the `axios` library.  
+Download it from https://github.com/opensearch-project/user-behavior-insights/tree/main/ubi-javascript-collector.
+
+Reference the events and create the client via:
+
+```js
+import { UbiEvent } from './ubi';
+import { UbiEventAttributes } from './ubi'
+import { UbiClient } from './ubi'
+
+const ubiClient = new  UbiClient('http://localhost:9200');
+```
+
+
+## Creating an event 
+
+This code snippet is to track adding an item to a shopping cart in an e-commerce application. It utilizes the `UbiEvent` and `UbiEventAttributes` class to encapsulate event details, which can then be sent to the tracking system.
+```js
+var event = new UbiEvent(
+    'add_to_cart', 
+    client_id, 
+    session_id, 
+    getQueryId(), 
+    new UbiEventAttributes('product', item.primary_ean, item.title, item), 
+    item.title + ' (' + item.id + ')'
+);
+```
+
+### Parameters
+
+1. **Event Name**: 
```

Comment:
Instead of a list, I would format this as a table with columns "Parameter", "Description", "Example".